src/Socket.php

```diff
@@ -36,16 +36,12 @@ public function __construct(LoopInterface $loop, $socket, Buffer $buffer = null)
 
     public function getLocalAddress()
     {
-        if ($this->socket !== false) {
-            return stream_socket_get_name($this->socket, false);
-        }
+        return $this->sanitizeAddress(@stream_socket_get_name($this->socket, false));
```

Comment:
Yes, i'm aware that `stream_socket_get_name`would return false for non-resources, but we're relying now on PHP triggering a warning.

I also noticed, that i looks like being more inconsistent as before.

* [`stream_socket_get_name` called without `@`](https://github.com/reactphp/socket/blob/master/src/Connection.php#L29).
* [`is_resource()` check](https://github.com/reactphp/socket/blob/master/src/Connection.php#L19)
* [`false` check](https://github.com/reactphp/datagram/blob/master/src/Socket.php#L39)

*Note, that i'm reviewing without too much knowledge about the internals, so my comments are not meant to block a merge.*